### PR TITLE
CXP-1298: Disable a catalog when removing a category that is in the product selection

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

https://user-images.githubusercontent.com/93598018/192547971-217011a1-3a1c-4a80-bc78-5ae1d0949b4b.mp4

- [x] Tests
- [x] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsContainingCategoryQuery.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+    public function __construct(
+        private Connection $connection,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(string $categoryCode): array
```
Comment (willy-ahva) [inline components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsContainingCategoryQuery.php:24]: There is something missing here. You have to handle all the children in the removed category. IMO you should add a new query like `getAllCildrenInCategory(string $categoryCode): array<string>`. Then, this query will be `execute(array $categoriesCode)` to handle several categories. I don't know how many parameters in the IN clause MySQL is able to handle. You may have to batch over Categories to apply this query with only subselection of codes.

File: components/catalogs/back/src/Infrastructure/Persistence/Category/GetAllCategoryCodesFromParentCategoryCodeQuery.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                UNION ALL
+                select
+                    p.id,
+                    p.code,
+                    p.parent_id
+                FROM pim_catalog_category p
+                INNER JOIN cte
+                ON p.parent_id = cte.id
+            )
+            SELECT code FROM cte
```
[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/src/Infrastructure/Persistence/Category/GetAllCategoryCodesFromParentCategoryCodeQuery.php:42]: That's exactly why we have `lft` and `rgt` in this table.

[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/src/Infrastructure/Persistence/Category/GetAllCategoryCodesFromParentCategoryCodeQuery.php:42]: Also, this is the first and only query in Catalogs where we look for another bounded context's data directly in SQL.
(The only exception is the JOIN between catalog & oro_user).
Maybe we can discuss with the griffins if we can help them add this to their new ServiceAPI: https://github.com/akeneo/pim-community-dev/tree/master/src/Akeneo/Category/back/ServiceApi

**Comment by guillaumefer-akeneo** [comment]
This PR is abandoned, the new simpler version will be in this branch : CXP-1306

**Comment by guillaumefer-akeneo** [comment]
This PR is abandoned, the new simpler version will be in this branch : CXP-1306

**Pull request closed without merging**